sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosDiagnosticsContext.java

```diff
@@ -392,23 +398,29 @@ synchronized void recordOperation(int statusCode, int subStatusCode, Integer act
             this.statusCode = statusCode;
             this.subStatusCode = subStatusCode;
             this.finalError = finalError;
-            this.actualItemCount += actualItemCount;
+            if (actualItemCount != null) {
+                if (this.actualItemCount.get() >= 0) {
+                    this.actualItemCount.addAndGet(actualItemCount);
+                } else {
+                    this.actualItemCount.set(actualItemCount);
```

Comment:
you can use:
if (!this.actualItemCount.compareAndSet(-1, actualItemCount)) {
  this.actualItemCount.addAndGet(actualItemCount);
}
It also does not suffer of multi-threading issue where 2 threads may end up on line 405

Same below on line 420.

Reply:
Done